### PR TITLE
ENH: Replaced csr_matrix() by tocsr() and complemented docstring

### PR DESCRIPTION
**Pull request by Didou09**
#### Reference issue
Issue #12249 
Reminder: small performance gain for `least_squares()` at every iteration when the user provides the jacobian as a sparse jacobian already in csr_matrix format

#### What does this implement/fix?
This PR:
* adds a parenthesis in least_squares docstr to mention that csr_matrix is preferred when jac is provided as a callable sparse.
* Replaces csr_matrix() instanciation by tocsr() method in the wrapper. As explained in issue #12249 , it should results in a x100 speed gain for that specific step in jacobian evaluation, for every evaluation

#### Additional information
Another option would have been to test J0 format and provide the wrapper only if it is not already in csr_matrix format, I leave it to the PR reviewer to judge which option is best.

### CODE REVIEW

File: scipy/optimize/_lsq/least_squares.py
Comment (perimosocordiae) [inline scipy/optimize/_lsq/least_squares.py:283]: typo: preferred

**Review by perimosocordiae — APPROVED** [review]
Looks good to me.

**Comment by Didou09** [comment]
Thanks for the positive review @perimosocordiae , 
* I see one of the Travis jobs failed, the message says it exceeded the maximum time limit for jobs, I guess it can be re-run ?
* The failed job on azure, it seems it was cancelled, (I have no idea why), can we re-run it too ? 

What is the next step after these tests have passed, before this PR can be merged ? Are other reviewers necessary ?

**Comment by tylerjereddy** [comment]
@Didou09  When you push in a fix for the typo CJ found it should flush through CI again, but yeah the issues look unrelated.

> What is the next step after these tests have passed, before this PR can be merged ? Are other reviewers necessary ?

No, one expert/core developer review is generally sufficient, so you're in good shape already! Maybe just fix that typo.

This is a performance enhancement PR, so I've adjusted the title of the PR to start with `ENH:` as is our custom. We [do the same for commit messages](https://numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message) when we can.

I'll add a milestone to this PR so we don't forget it before the 1.6.0 release, but looks like it will get merged long before then.

**Comment by Didou09** [comment]
Thanks for your answer @tylerjereddy ,

I committed the typo correction, the tests re-ran, it looks like 2 travis tests failed:
* one due to [exceeded time limit](https://travis-ci.org/github/scipy/scipy/jobs/692442629)
* one seems to highlight a [real test failure](https://travis-ci.org/github/scipy/scipy/jobs/692442632), in the lapack library, but I'm not sure it is related to the changes I proposed, an expert's informed opinion seems necessary:
```
=================================== FAILURES ===================================

_________________________________ test_heequb __________________________________

[gw2] linux -- Python 3.7.5 /home/travis/build/scipy/scipy/builds/venv/bin/python

scipy/linalg/tests/test_lapack.py:1759: in test_heequb

    assert_allclose(np.log2(s), [0., -1.]*2 + [0.] + [-4]*5)

E   AssertionError: 

E   Not equal to tolerance rtol=1e-07, atol=0

E   

E   Mismatched elements: 7 / 10 (70%)

E   Max absolute difference: 5.

E   Max relative difference: 1.25

E    x: array([ 0.,  0.,  0.,  0.,  0., -9., -9., -9., -9., -9.])

E    y: array([ 0., -1.,  0., -1.,  0., -4., -4., -4., -4., -4.])

        A          = array([[   2.+0.j,    0.+1.j,    0.+0.j,    0.+0.j,    0.+0.j,    0.+0.j,

           0.+0.j,    0.+0.j,    0.+0.j,    ...   0.+0.j,    0.+0.j,    0.+0.j,    0.+0.j,    0.+0.j,    0.+0.j,

           0.+0.j,    0.+0.j,    0.+0.j, 1002.+0.j]])

        amax       = 1002.0

        info       = 0

        s          = array([1.        , 1.        , 1.        , 1.        , 1.        ,

       0.00195312, 0.00195312, 0.00195312, 0.00195312, 0.00195312])

        scond      = 0.001953125
```

**Comment by perimosocordiae** [comment]
Yeah, the CI failures are unrelated to this change. Merged, thanks @Didou09 !

**Pull request merged**
